### PR TITLE
Tiny state management improvements

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

Part of #8785 

- Typo fixes
- Simplify implementation of `markFieldAsDirty`
- Micro optimization to `addDirtyIndices`
- Remove redundant dirty indices setter in `rootSelector`

### CODE REVIEW

File: beacon-chain/state/v1/state_trie.go [hunk 1]
```diff
@@ -410,7 +410,6 @@ func (b *BeaconState) rootSelector(ctx context.Context, field fieldIndex) ([32]b
 			if err != nil {
 				return [32]byte{}, err
 			}
-			b.dirtyIndices[field] = []uint64{}
```
Comment (rkapka) [inline beacon-chain/state/v1/state_trie.go:413]: This is covered in `b.resetFieldTrie`

File: beacon-chain/state/v1/field_trie_helpers.go [hunk 2]
```diff
@@ -27,8 +27,8 @@ func validateElements(field fieldIndex, elements interface{}, length uint64) err
 	return nil
 }
 
-// this converts the corresponding field and the provided elements to the appropriate roots.
-func fieldConverters(field fieldIndex, indices []uint64, elements interface{}, convertAll bool) ([][32]byte, error) {
+// convertToRoots converts the corresponding field and the provided elements to the appropriate roots.
+func convertToRoots(field fieldIndex, indices []uint64, elements interface{}, convertAll bool) ([][32]byte, error) {
```
Comment (nisdas) [inline beacon-chain/state/v1/field_trie_helpers.go:31]: don't think `convertToRoots` is a particularly good name for what this method actually does, it is too generic a name here.

File: beacon-chain/state/v1/setters_misc.go [hunk 3]
```diff
@@ -175,9 +171,10 @@ func (b *BeaconState) addDirtyIndices(index fieldIndex, indices []uint64) {
 	if b.rebuildTrie[index] {
 		return
 	}
-	b.dirtyIndices[index] = append(b.dirtyIndices[index], indices...)
-	if len(b.dirtyIndices[index]) > indicesLimit {
+	if len(b.dirtyIndices[index])+len(indices) > indicesLimit {
```
Comment (nisdas) [inline beacon-chain/state/v1/setters_misc.go:174]: can you declare this as a separate var
```
len(b.dirtyIndices[index])+len(indices)
```

File: beacon-chain/state/v1/field_trie_helpers.go [hunk 4]
```diff
@@ -27,8 +27,8 @@ func validateElements(field fieldIndex, elements interface{}, length uint64) err
 	return nil
 }
 
-// this converts the corresponding field and the provided elements to the appropriate roots.
-func fieldConverters(field fieldIndex, indices []uint64, elements interface{}, convertAll bool) ([][32]byte, error) {
+// convertFieldElementsToRoots converts the corresponding field and the provided elements to the appropriate roots.
+func convertFieldElementsToRoots(field fieldIndex, indices []uint64, elements interface{}, convertAll bool) ([][32]byte, error) {
```
Comment (nisdas) [inline beacon-chain/state/v1/field_trie_helpers.go:31]: this now seems too verbose, how about we just leave it as it is ?

Comment (rkapka) [reply beacon-chain/state/v1/field_trie_helpers.go:31]: ok

**Review by nisdas — APPROVED** [review]
LGTM

**Pull request merged**